2.0/s2i/bin/run

```diff
@@ -5,20 +5,39 @@ function is_puma_installed() {
   grep ' puma ' Gemfile.lock >/dev/null
 }
 
+function is_rerun_installed() {
+  [ ! -f Gemfile.lock ] && return 1
+  grep ' rerun ' Gemfile.lock >/dev/null
+}
+
 set -e
 
 export RACK_ENV=${RACK_ENV:-"production"}
 
+if [[ "$RACK_ENV" == "development" ]]
+  if is_rerun_installed; then
+    export RERUN="rerun --"
+  else
+    echo "NOTICE: Your application is running in 'development' environment"
+    echo "        but your Gemfile is missing dependencies for hot deploy."
+    echo "        In order to dynamically pick up changes made in your"
+    echo "        source code, add following line to your Gemfile:"
+    echo
+    echo "        gem 'rerun'"
+    echo
+  fi
+fi
```

Comment:
I thought that we have separate s2i scripts for the rails-ex, since we strip the ror stuff from the sti-ruby, but no I see we need it here.